alvin_c/server.js

```diff
@@ -0,0 +1,108 @@
+var express = require('express');
+var app = express();
+//
+var bodyParser = require('body-parser');
+app.use(bodyParser.urlencoded({ extended: true }));
+
+var path = require('path');
+app.use(express.static(path.join(__dirname, './static')));
+app.set('views', path.join(__dirname, './views'));
+
+app.set('view engine', 'ejs');
+
+// import and use mongoose
+var mongoose = require('mongoose');
+mongoose.connect('mongodb://localhost/basic_meese');
+var MeeseSchema = new mongoose.Schema({
+ name: String,
+ age: Number
+})
+mongoose.model('Meese', MeeseSchema); // We are setting this Schema in our Models as 'meese'
+var Meese = mongoose.model('Meese') // We are retrieving this Schema from our Models, named 'meese'
+mongoose.Promise = global.Promise; // because reasons
+
+// Routes //
+// display all of the meese
+app.get('/', function(req, res) {
+  Meese.find({}, function(err, meese){
+    if(!err){
+      //console.log(meese);
+      res.render('index',{meese:meese});
+    }
+    else {
+      console.log('database find error occured');
+      res.send('error!');
+    }
+  })
+})
+// displays information about one meese
+app.get('/meese/new', function(req, res){
+  res.render('form');
+})
+// displays a form for making a new meese
+app.get('/meese/:id', function(req, res){
+  console.log(req.params.id);
+  Meese.find({_id:req.params.id}, function(err, meese){
+    if(!err){
+      //console.log(meese);
+      res.render('profile',{meese:meese});
+    }
+    else {
+      console.log('database find error occured');
+      res.send('error!');
+    }
+  })
+})
+// should show a form to edit an existing meese
+app.get('/meese/:id/edit', function(req, res){
+  console.log('edit path called'+req.params.id)
+  Meese.find({_id:req.params.id}, function(err, meese){
+    if(!err){
+      //console.log(meese);
+      res.render('form1',{meese:meese});
```

Comment:
Ideally, the view for a new meese would be called 'new.ejs'. Remember, your view names will be available on the client, so every user of your site will be able to read 'form1' through inspect element.